external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -95,15 +112,51 @@ private[kafka010] class KafkaMicroBatchStream(
   override def latestOffset(start: Offset, readLimit: ReadLimit): Offset = {
     val startPartitionOffsets = start.asInstanceOf[KafkaSourceOffset].partitionToOffsets
     latestPartitionOffsets = kafkaOffsetReader.fetchLatestOffsets(Some(startPartitionOffsets))
+    var skipBatch = false
     endPartitionOffsets = KafkaSourceOffset(readLimit match {
+      case rows: ReadMinRows =>
+        // checking if we need to skip batch based on minOffsetPerTrigger criteria
+        skipBatch = delayBatch(rows.minRows(), latestPartitionOffsets, startPartitionOffsets)
+        if (skipBatch) {
+          logDebug(
+            s"Delaying batch as number of records available is less than minOffsetsPerTrigger")
+          startPartitionOffsets
+        } else latestPartitionOffsets
       case rows: ReadMaxRows =>
         rateLimit(rows.maxRows(), startPartitionOffsets, latestPartitionOffsets)
+      case rows: CompositeReadLimit =>
+        skipBatch = delayBatch(rows.minRows(), latestPartitionOffsets, startPartitionOffsets)
+        if (skipBatch) {
+          logDebug(
+            s"Delaying batch as number of records available is less than minOffsetsPerTrigger")
+          startPartitionOffsets
+        } else {
+          rateLimit(rows.maxRows(), startPartitionOffsets, latestPartitionOffsets)
+        }
       case _: ReadAllAvailable =>
         latestPartitionOffsets
     })
+    if (!skipBatch) lastTriggerMillis = System.currentTimeMillis()
     endPartitionOffsets
   }
 
+  /** Checks if we need to skip this trigger based on minOffsetsPerTrigger & maxTriggerDelay */
+  private def delayBatch(minLimit: Long,
```

Comment:
Indentation looks to be off: please refer the style guide on indentation of the method parameters.
https://github.com/databricks/scala-style-guide